api-specificatie/BRP-Wijzigingen/openapi.yaml

```diff
@@ -0,0 +1,146 @@
+openapi: 3.0.0
+info:
+  title: "BRP Wijzigingen API"
+  version: "1.0.0"
+  contact:
+    name: test
+    email: me@home.nl
+    url: http://home.nl
+paths:
+  /wijzigingen:
+    get:
+      tags: 
+      - wijzigingen
+      operationId: getWijzigingen
+      parameters:
+        - name: afnemersindicatie
+          in: query
+          description: afnemersindicatie
+          required: true
+          schema:
+            type: string
+        - name: van
+          in: query
+          description: "van (yyyy-mm-ddThh:mm:ss)"
+          required: true
+          schema:
+            type: string
+            format: datetime
+        - name: tot
+          in: query
+          description: "tot (yyyy-mm-ddThh:mm:ss)"
+          required: false
+          schema:
+            type: string
+            format: datetime
+      responses:
+        200:
+          description: "response"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WijzigingenHal'
+              examples:
+                gewijzigdIngeschrevenPersoon:
+                  $ref: '#/components/examples/GewijzigdIngeschrevenPersoon'
+                langerDanEenWeek:
+                  $ref: '#/components/examples/BevraagPeriodeLangerDanEenWeek'
+                alleenVan:
+                  $ref: '#/components/examples/AlleenVanDatumTijd'
+        400:
+          $ref: '#/components/responses/400'
+
+components:
+  responses:
+    '400':
+      description: Bad Request
+      content:
+        application/problem+json:
+          schema:
+            $ref: 'https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/master/api-specificatie/common.yaml#/components/schemas/BadRequestFoutbericht'
+          examples:
+            alleenTotDatum:
+              $ref: '#/components/examples/AlleenTotDatumTijd'
+
+  schemas:
+    WijzigingenHal:
+      type: object
+      properties:
+        _links:
+          $ref: '#/components/schemas/WijzigingenHal_links'
+        wijzigingen:
```

Comment:
Ik heb dit ook in de BRK Update API gedaan, ik weet alleen niet of het in een _embedded moet worden gestopt om het consistent te maken met de bevragen API's

Reply:
Ah ja, je bedoelt of je naast de _links property ook een _embedded property moet hebben. Het zou dan zoiets worden als:

```
"""
    {
        "_links": {
            "self": { "href": "/wijzigingen?van=2020-05-11T14:00:00&tot=2020-05-11T16:00:00&afnemerindicatie=085801" },
            "ingeschrevenPersoon": { "href": "/ingeschrevenpersonen/{burgerservicenummer}", "templated": true }
        },
        "_embedded": {
            "wijzigingen": [
                {
                    "burgerservicenummer": "999999321",
                    "type": "ingeschrevenpersoon",
                    "mutatietijdstip": "2020-05-11T14:13:37"
                },
                {
                    "burgerservicenummer": "999999322",
                    "type": "ingeschrevenpersoon",
                    "mutatietijdstip": "2020-05-11T14:46:49"
                },
                {
                    "burgerservicenummer": "999999323",
                    "type": "ingeschrevenpersoon",
                    "mutatietijdstip": "2020-05-11T14:58:12"
                }
            ]
        }
    }
    """
```

Het lijkt dan inderdaad consistent te zijn met de bevragen API's. De vraag is alleen of je de individuele wijzigingen in het bovenstaande bericht als aparte resources kunt beschouwen. In dat geval zijn het hier subresources en zou je volgens HAL een _embedded op moeten nemen.
Wanneer spreek je echter van een resource? Moet het dan niet ook een identificatie hebben?
Zo ja, hebben bovenstaande wijzigingen identificaties?

Reply:
Volgens mij is wat je nu aangeeft de reden waarom bij de BRK Update API er voor is gekozen om de wijzigingen niet in _embedded te stoppen. Een wijziging is geen resource

Reply:
Die conclusie had ik voor mezelf eigenlijk ook al genomen. Ik stel dus voor om de yaml op dit punt te laten voor wat hij is.